src/schema_aggregator/user-interface/site-schema-response-header-integration.php

```diff
@@ -0,0 +1,75 @@
+<?php
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong -- Needed in the folder structure.
+namespace Yoast\WP\SEO\Schema_Aggregator\User_Interface;
+
+use WP_REST_Request;
+use WP_REST_Response;
+use WPSEO_Utils;
+use Yoast\WP\SEO\Conditionals\No_Conditionals;
+use Yoast\WP\SEO\Integrations\Integration_Interface;
+
+/**
+ * Integration to set proper response headers for Site Schema endpoints.
+ */
+class Site_Schema_Response_Header_Integration implements Integration_Interface {
```

Comment:
I feel this should live in the `infrastructure` layer: in the end HTTP headers are infrastructural concerns, right? Maybe this should be an adapter of some sort...